svf/lib/Util/CallGraphBuilder.cpp

```diff
@@ -56,28 +54,25 @@ CallGraph* CallGraphBuilder::buildCallGraph(SVFModule* svfModule)
                     const CallICFGNode* callBlockNode = cast<CallICFGNode>(inst);
                     if(const SVFFunction* callee = callBlockNode->getCalledFunction())
                     {
-                        callgraph->addDirectCallGraphEdge(callBlockNode,*F,callee);
+                        callgraph->addDirectCallGraphEdge(callBlockNode,(item.second)->getFunction(),callee);
                     }
                 }
             }
         }
     }
-
     return callgraph;
 }
 
-CallGraph* ThreadCallGraphBuilder::buildThreadCallGraph(SVFModule* svfModule)
+CallGraph* CallGraphBuilder::buildThreadCallGraph()
 {
-
-    buildCallGraph(svfModule);
-
+    CallGraph* callgraph = new ThreadCallGraph(*(PAG::getPAG()->getCallGraph()));
     ThreadCallGraph* cg = dyn_cast<ThreadCallGraph>(callgraph);
     assert(cg && "not a thread callgraph?");
 
     ThreadAPI* tdAPI = ThreadAPI::getThreadAPI();
-    for (SVFModule::const_iterator F = svfModule->begin(), E = svfModule->end(); F != E; ++F)
+    for (const auto& item: *PAG::getPAG()->getCallGraph())
```

Comment:
use a tmp variable to store PAG::getPAG()->getCallGraph() at the beginning of the function